MediaWikiKit/MediaWikiKit/MWKRecentSearchList.m

```diff
@@ -37,22 +38,28 @@ - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
     return self;
 }
 
+#pragma mark - Validation
+
+- (BOOL)isEntryValid:(MWKRecentSearchEntry*)entry {
+    return entry.searchTerm.length > 0 && entry.site;
+}
+
 #pragma mark - Data Update
 
+- (void)importEntries:(NSArray*)entries {
+    [super importEntries:[entries bk_select:^BOOL (MWKRecentSearchEntry* entry) {
+        return [self isEntryValid:entry];
+    }]];
+}
+
 - (void)addEntry:(MWKRecentSearchEntry*)entry {
-    if (entry.searchTerm == nil) {
+    if (![self isEntryValid:entry]) {
```

Comment:
Should this be a super class method? 

I guess this is related to the other comment… are all the lists validating data differently?